nx/lib/nx/lin_alg.ex

```diff
@@ -1883,4 +2076,27 @@ defmodule Nx.LinAlg do
     # Calculate matrix rank
     Nx.sum(s > tol)
   end
+
+  defp apply_vectorized(tensor, fun) when is_function(fun, 1) do
```

Comment:
Shouldn't we also write our own `apply_vectorized` in a way similar to this?

Reply:
I believe we can!
Originally we didn't have the proper tools in place :)